src/game_battlealgorithm.cpp

```diff
@@ -1190,7 +1190,8 @@ bool Game_BattleAlgorithm::Skill::IsTargetValid() const {
 		skill.scope == RPG::Skill::Scope_party) {
 		if (GetTarget()->IsDead()) {
 			// Cures death
-			return !skill.state_effects.empty() && skill.state_effects[0];
+			return !skill.state_effects.empty() && skill.state_effects[0] &&
+				!(Player::IsRPG2k3() && skill.state_effect);
 		}
```

Comment:
I'd actually like to see this feature as an EasyRPG extension to 2k battle system. But that can be added later.